runtime/src/snapshot_utils.rs

```diff
@@ -1118,24 +1134,34 @@ where
         .prefix(unpacked_snapshots_dir_prefix)
         .tempdir_in(bank_snapshots_dir)?;
     let unpacked_snapshots_dir = unpack_dir.path().join("snapshots");
+    let unpacked_version_file = unpack_dir.path().join("version");
 
-    let mut measure_untar = Measure::start(measure_name);
-    let unpacked_append_vec_map = untar_snapshot_in(
-        snapshot_archive_path,
-        unpack_dir.path(),
-        account_paths,
+    let file_receiver = SnapshotUnpacker::spawn_unpack_snapshot(
+        account_paths.to_vec(),
+        unpack_dir.path().to_path_buf(),
+        snapshot_archive_path.as_ref().to_path_buf(),
         archive_format,
         parallel_divisions,
-    )?;
-    measure_untar.stop();
+    );
+    let num_rebuilder_threads = std::cmp::max(
+        1,
+        num_cpus::get_physical().saturating_sub(parallel_divisions),
```

Comment:
We may want to decouple these tuning numbers. `parallel_divisions` is a different tunable parameter than how many threads we want dealing with append vecs. `parallel_divisions` is probably? based on disk i/o while untarring. I don't know what the rebuilder threads will be limited by. This can be for later.

Reply:
and I learned a new trick lately. `1.max(num_cpus::get_physical().saturating_sub(parallel_divisions))` is more concise.

Reply:
or probably better the other way
`num_cpus::get_physical().saturating_sub(parallel_divisions).max(1)`